app/src/main/java/com/example/househomey/UserProfileFragment.java

```diff
@@ -0,0 +1,75 @@
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+public class UserProfileFragment extends Fragment {
+
+    private String username;
+    TextView usernameTextView;
+    Button logoutButton;
+
+    /**
+     * Called to create the user profile fragment's view hierarchy. Inflates the fragment layout from
+     * the specified XML resource, populates the item list from the bundle arguments, and sets up UI
+     * elements such as the username TextView and logout button.
+     *
+     * @param inflater           The LayoutInflater object that can be used to inflate any views in
+     *                           the fragment.
+     * @param container          If non-null, this is the parent view that the fragment's UI should be
+     *                           attached to. The fragment should not add the view itself, but this can
+     *                           be used to generate the LayoutParams of the view.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed from a previous
+     *                           saved state as given here.
+     * @return The root view of the fragment's layout hierarchy.
+     */
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+
+        // Inflate the fragment's layout
+        View rootView = inflater.inflate(R.layout.fragment_user_profile, container, false);
+
+        //Populate the item list from bundle
+        Bundle args = getArguments();
+
+        this.username = args.getString("username");
+
+        usernameTextView = rootView.findViewById(R.id.user_profile_username);
+        logoutButton = rootView.findViewById(R.id.user_profile_logout_button);
+
+        usernameTextView.setText(this.username);
+        logoutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FirebaseAuth.getInstance().signOut();
+                logoutButton.setBackgroundResource(R.drawable.logout_button_clicked);
+                logoutButton.setTextColor(getResources().getColor(R.color.creme, rootView.getContext().getTheme()));
+                new Handler(Looper.getMainLooper()).postDelayed(() -> {
+                    logoutButton.setBackgroundResource(R.drawable.logout_button_unclicked);
+                    logoutButton.setTextColor(getResources().getColor(R.color.black, rootView.getContext().getTheme()));
+                }, 100);
+                Activity activity = getActivity();
+                if (activity != null) {
+                    Intent intent = new Intent(activity, SignInActivity.class);
+                    activity.startActivity(intent);
```

Comment:
We need to call ` activity.finish()` after starting each new activity. 

Otherwise, it's possible to get back into the previous activity (after logging out) by swiping left on the screen (which simulates Android's back button)

Give it a try after logging out, swipe left